mev-boost-rs/src/service.rs

```diff
@@ -53,12 +53,17 @@ pub struct Service {
 
 impl Service {
     pub fn from(config: Config) -> Self {
-        let relays: Vec<RelayEndpoint> = config
-            .relays
-            .iter()
-            .filter_map(|s| parse_url(s))
-            .filter_map(|url| RelayEndpoint::try_from(url).ok())
-            .collect();
+        let mut relays: Vec<RelayEndpoint> = Vec::new();
+
+        for s in &config.relays {
+            match parse_url(s) {
+                Some(url) => match RelayEndpoint::try_from(url) {
+                    Ok(relay) => relays.push(relay),
+                    Err(_) => tracing::warn!("Failed to parse relay endpoint: {}", s),
+                },
+                None => tracing::warn!("Failed to parse URL: {}", s),
+            }
+        }
```

Comment:
1. use more descriptive symbol for relay url than just `s`
2. and move this code to the `parse_url` fn and rename to reflect the expanded logic

```suggestion
        for relay_url in &config.relays {
            match input.parse::<Url>() {
                Ok(url) => match RelayEndpoint::try_from(url) {
                    Ok(relay) => relays.push(relay),
                    Err(err) => tracing::warn!("error parsing relay endpoint `{url}`: {err}"),
                },
                Err(err) => tracing::warn!("error parsing relay from URL `{relay_url}`: {err}"),
            }
        }
```

so `parse_url(input: &str) -> Option<Url>` becomes something like `parse_relay_endpoints(relay_urls: &[String]) -> Vec<RelayEndpoint>`